mlir/lib/Dialect/Tosa/Transforms/TosaFolders.cpp

```diff
@@ -1322,10 +1322,14 @@ DenseElementsAttr padType(ShapedType inputType, ElementsAttr inputValues,
         llvm::any_of(llvm::enumerate(indexInTarget), [&](const auto &dimInfo) {
```

Comment:
It's odd to mutate indexInTarget within any_of, which should be used as a constant function to determine an attribute.
How about we split this into
`indexInTarget  = indexInTarget - padding[*][0]` (encapsulated into llvm::for_each + llvm::enumerate() )
followed by `isPad = any_of(indexInTarget) <0 || || i >= inputShape[index])`
?